In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sys

visa=pd.read_csv(r"C:\Users\ADMIN\OneDrive\Documents\data science nit\Pandas\Visadataset.csv")
visa

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


**step-2:seperate cat and num columns**

In [5]:
cat=visa.select_dtypes(include='object').columns.tolist()
num=visa.select_dtypes(exclude='object').columns.tolist()
cat,num

(['case_id',
  'continent',
  'education_of_employee',
  'has_job_experience',
  'requires_job_training',
  'region_of_employment',
  'unit_of_wage',
  'full_time_position',
  'case_status'],
 ['no_of_employees', 'yr_of_estab', 'prevailing_wage'])

- frequency table
- relative frequency table
- bar chart
- pie chart

**Step-3:Frequency Table**

In [6]:
cat

['case_id',
 'continent',
 'education_of_employee',
 'has_job_experience',
 'requires_job_training',
 'region_of_employment',
 'unit_of_wage',
 'full_time_position',
 'case_status']

In [7]:
visa['continent']

0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

In [8]:
visa['continent'].unique()

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [9]:
visa['continent'].nunique()

6

- There are 25480 applicants from  6 continent
- number of application from each continent:**Frequency table**

In [10]:
visa      #df
visa['continent']  # one column
con=visa['continent']=='Asia'  # one class
np.sum(con)   #sum
visa[con]     #extraction

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
5,EZYV06,Asia,Master's,Y,N,2339,2012,South,78252.1400,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


In [11]:
# con=visa['continent']=='Asia'  # one class
# np.sum(con)

# con=visa['continent']=='Africa'  # one class
# np.sum(con)


# con_class=visa['continent']=='continent'  # one class
# np.sum(con)
cont_class=visa['continent'].unique()
freq=[]
for i in cont_class:
    con=visa['continent']==i
    freq.append(np.sum(con))

pd.DataFrame(zip(cont_class,freq),
            columns=['Class','Frequency'])
            

,Class,Frequency
0,Asia,16861
1,Africa,551
2,North America,3292
3,Europe,3732
4,South America,852
5,Oceania,192


In [12]:
visa['continent'].value_counts()


continent
Asia             16861
Europe            3732
North America     3292
South America      852
Africa             551
Oceania            192
Name: count, dtype: int64

In [13]:
visa['continent'].keys()

RangeIndex(start=0, stop=25480, step=1)

In [14]:
visa['continent'].values

array(['Asia', 'Asia', 'Asia', ..., 'Asia', 'Asia', 'Asia'],
      shape=(25480,), dtype=object)

In [15]:
keys=visa.keys().tolist()
values=visa.dtypes.values.tolist()
cat,num=[],[]
for i,j in zip(keys,values):
    if j=='object':
        cat.append(i)
    else:
        num.append(i)
cat,num

(['case_id',
  'continent',
  'education_of_employee',
  'has_job_experience',
  'requires_job_training',
  'region_of_employment',
  'unit_of_wage',
  'full_time_position',
  'case_status'],
 ['no_of_employees', 'yr_of_estab', 'prevailing_wage'])

In [16]:
visa=pd.read_csv(r"C:\Users\ADMIN\OneDrive\Documents\data science nit\Pandas\Visadataset.csv")

keys=visa['continent'].value_counts().keys()
values=visa['continent'].value_counts().values
pd.DataFrame(zip(keys,values),
            columns=['continent','Frequency'])

,continent,Frequency
0,Asia,16861
1,Europe,3732
2,North America,3292
3,South America,852
4,Africa,551
5,Oceania,192


In [17]:
visa.groupby(by='continent').count()['case_id']

continent
Africa             551
Asia             16861
Europe            3732
North America     3292
Oceania            192
South America      852
Name: case_id, dtype: int64

In [18]:
visa.groupby(by='continent').count()['education_of_employee']

continent
Africa             551
Asia             16861
Europe            3732
North America     3292
Oceania            192
South America      852
Name: education_of_employee, dtype: int64

In [19]:
visa.groupby(by='education_of_employee').count()['case_status']

education_of_employee
Bachelor's     10234
Doctorate       2192
High School     3420
Master's        9634
Name: case_status, dtype: int64

In [20]:
visa.groupby(by='no_of_employees').count()['has_job_experience']

no_of_employees
-26        7
-25        5
-17        4
-14        8
-11        9
          ..
 547172    1
 579004    1
 581468    1
 594472    1
 602069    1
Name: has_job_experience, Length: 7105, dtype: int64

In [21]:
# cont_class=visa['continent'].unique()
# freq=[]
# for i in cont_class:
#     con=visa['continent']==i
#     freq.append(np.sum(con))

# pd.DataFrame(zip(cont_class,freq),
#             columns=['Class','Frequency'])
cont_class=visa['region_of_employment'].unique()
freq=[]
for i in cont_class:
    con=visa['region_of_employment']==i
    freq.append(np.sum(con))
pd.DataFrame(zip(cont_class,freq),
            columns=['class','frequency'])

,class,frequency
0,West,6586
1,Northeast,7195
2,South,7017
3,Midwest,4307
4,Island,375


In [22]:
cont_class=visa['case_id'].unique()
freq=[]
for i in cont_class:
    con=visa['case_id']==i
    freq.append(np.sum(con))
pd.DataFrame(zip(cont_class,freq),
            columns=['class','frequency'])


,class,frequency
0,EZYV01,1
1,EZYV02,1
2,EZYV03,1
3,EZYV04,1
4,EZYV05,1
...,...,...
25475,EZYV25476,1
25476,EZYV25477,1
25477,EZYV25478,1
25478,EZYV25479,1


In [23]:
cont_class=visa['case_id'].unique()
freq=[]
for i in cont_class:
    con=visa['case_id']==i
    freq.append(np.sum(con))
pd.DataFrame(zip(cont_class,freq),
            columns=['class','frequency'])


,class,frequency
0,EZYV01,1
1,EZYV02,1
2,EZYV03,1
3,EZYV04,1
4,EZYV05,1
...,...,...
25475,EZYV25476,1
25476,EZYV25477,1
25477,EZYV25478,1
25478,EZYV25479,1


In [24]:
cont_class=visa['case_id'].unique()
freq=[]
for i in cont_class:
    con=visa['case_id']==i
    freq.append(np.sum(con))
pd.DataFrame(zip(cont_class,freq),
            columns=['class','frequency'])


,class,frequency
0,EZYV01,1
1,EZYV02,1
2,EZYV03,1
3,EZYV04,1
4,EZYV05,1
...,...,...
25475,EZYV25476,1
25476,EZYV25477,1
25477,EZYV25478,1
25478,EZYV25479,1


In [25]:
visa

,case_id,continent,education_of_employee,has_job_experience,requires_job_training,no_of_employees,yr_of_estab,region_of_employment,prevailing_wage,unit_of_wage,full_time_position,case_status
0,EZYV01,Asia,High School,N,N,14513,2007,West,592.2029,Hour,Y,Denied
1,EZYV02,Asia,Master's,Y,N,2412,2002,Northeast,83425.6500,Year,Y,Certified
2,EZYV03,Asia,Bachelor's,N,Y,44444,2008,West,122996.8600,Year,Y,Denied
3,EZYV04,Asia,Bachelor's,N,N,98,1897,West,83434.0300,Year,Y,Denied
4,EZYV05,Africa,Master's,Y,N,1082,2005,South,149907.3900,Year,Y,Certified
...,...,...,...,...,...,...,...,...,...,...,...,...
25475,EZYV25476,Asia,Bachelor's,Y,Y,2601,2008,South,77092.5700,Year,Y,Certified
25476,EZYV25477,Asia,High School,Y,N,3274,2006,Northeast,279174.7900,Year,Y,Certified
25477,EZYV25478,Asia,Master's,Y,N,1121,1910,South,146298.8500,Year,N,Certified
25478,EZYV25479,Asia,Master's,Y,Y,1918,1887,West,86154.7700,Year,Y,Certified


In [27]:
cont_class=visa['case_id'].unique()
freq=[]
for i in cont_class:
    con=visa['case_id']==i
    freq.append(np.sum(con))
pd.DataFrame(zip(cont_class,freq),
            columns=['class','frequency'])
freq


[np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.int64(1),
 np.in

In [2]:
print('hello')

hello
